django_extensions/management/shells.py

```diff
@@ -9,64 +9,90 @@ class ObjectImportError(Exception):
     pass
 
 
+def check_alias(directive):
+    result = directive.split(' as ')
+    if len(result) == 2:
+        object_name, alias = result
+    else:
+        object_name = alias = result[0]
+    return object_name, alias
+
+
 def import_items(import_directives, style, quiet_load=False):
     """
     Import the items in import_directives and return a list of the imported items
 
     Each item in import_directives should be one of the following forms
-        * a tuple like ('module.submodule', ('classname1', 'classname2')), which indicates a 'from module.submodule import classname1, classname2'
-        * a tuple like ('module.submodule', 'classname1'), which indicates a 'from module.submodule import classname1'
-        * a tuple like ('module.submodule', '*'), which indicates a 'from module.submodule import *'
-        * a simple 'module.submodule' which indicates 'import module.submodule'.
+        * a tuple like ('module.submodule', ('classname1', 'classname2 as alias')), which indicates a 'from module.submodule import classname1, classname2 as alias'
+        * a tuple like ('module.submodule', 'classname1[ as alias]'), which indicates 'from module.submodule import classname1[ as alias]'
+        * a tuple like ('module.submodule', '*'), which indicates 'from module.submodule import *'
+        * a string like 'module.submodule[ as alias]', which indicates 'import module.submodule[ as alias]'.
 
-    Returns a dict mapping the names to the imported items
+    Returns a dict mapping the names or aliases to the imported items
     """
     imported_objects = {}
     for directive in import_directives:
         try:
             # First try a straight import
             if isinstance(directive, six.string_types):
-                imported_object = __import__(directive)
-                imported_objects[directive.split('.')[0]] = imported_object
+                object_name, alias = check_alias(directive)
+                if object_name == alias:
```

Comment:
I wanted to avoid having to do 
```
if alias: 
    imported_objects[alias] = imported_object
else:
    imported_objects[object_name] = imported_object
```
after each call to `check_alias`, though I think the "check_alias" method name could be better. Maybe using "name" as the variable name instead of "alias" would be more conceptually clear?